components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/ApplicationImportExportManager.java

```diff
@@ -134,28 +134,56 @@ public boolean isOwnerAvailable(String userId) throws APIManagementException {
-                    API api = apiSet.iterator().next();
-                    //tier of the imported subscription
-                    Tier tier = subscribedAPI.getTier();
-                    //checking whether the target tier is available
-                    if (isTierAvailable(tier, api) && api.getStatus() != null &&
-                            APIConstants.PUBLISHED.equals(api.getStatus())) {
-                        ApiTypeWrapper apiTypeWrapper = new ApiTypeWrapper(api);
-                        apiTypeWrapper.setTier(tier.getName());
-                        // add subscription if update flag is not specified
-                        // it will throw an error if subscriber already exists
-                        if (update == null || !update) {
-                            apiConsumer.addSubscription(apiTypeWrapper, userId, appId);
-                        } else if (!apiConsumer.isSubscribedToApp(subscribedAPI.getApiId(), userId, appId)) {
-                            // on update skip subscriptions that already exists
-                            apiConsumer.addSubscription(apiTypeWrapper, userId, appId);
+                    Object type = apiSet.iterator().next();
+                    //Check whether the object is an ApiProduct
+                    if (isApiProduct(type)) {
+                        //Handle Api Product subscriptions
+                        APIProduct apiProduct = (APIProduct) apiSet.iterator().next();
+                        //tier of the imported subscription
+                        Tier tier = subscribedAPI.getTier();
+                        //checking whether the target tier is available
+                        if (isTierAvailableForProduct(tier, apiProduct) && apiProduct.getState() != null &&
+                                APIConstants.PUBLISHED.equals(apiProduct.getState())) {
+                            ApiTypeWrapper apiTypeWrapper = new ApiTypeWrapper(apiProduct);
+                            apiTypeWrapper.setTier(tier.getName());
+                            // add subscription if update flag is not specified
+                            // it will throw an error if subscriber already exists
+                            if (update == null || !update) {
+                                apiConsumer.addSubscription(apiTypeWrapper, userId, appId);
+                            } else if (!apiConsumer.isSubscribedToApp(subscribedAPI.getApiId(), userId, appId)) {
+                                // on update skip subscriptions that already exists
+                                apiConsumer.addSubscription(apiTypeWrapper, userId, appId);
+                            }
+                        } else {
+                            log.error("Failed to import Subscription as API " + name + "-" + version +
+                                    " as one or more tiers may be unavailable or the API may not have been published ");
+                            skippedAPIList.add(subscribedAPI.getApiId());
```

Comment:
I think we can remove the duplicated code segments as below. Just check and do the required changes.

```suggestion
                    ApiTypeWrapper apiTypeWrapper = null;
                    if (isApiProduct(type)) {
                        APIProduct apiProduct = (APIProduct) apiSet.iterator().next();
                        apiTypeWrapper = new ApiTypeWrapper(apiProduct);
                    } else {
                        API api = (API) apiSet.iterator().next();
                        apiTypeWrapper = new ApiTypeWrapper(api);
                    }
                    //tier of the imported subscription
                    ier tier = subscribedAPI.getTier();
                    //checking whether the target tier is available
                    if (isTierAvailable(tier, apiTypeWrapper) && apiTypeWrapper.getStatus() != null &&
                            APIConstants.PUBLISHED.equals(apiTypeWrapper.getStatus())) {
                            apiTypeWrapper.setTier(tier.getName());
                            // add subscription if update flag is not specified
                            // it will throw an error if subscriber already exists
                            if (update == null || !update) {
                                apiConsumer.addSubscription(apiTypeWrapper, userId, appId);
                            } else if (!apiConsumer.isSubscribedToApp(subscribedAPI.getApiId(), userId, appId)) {
                                // on update skip subscriptions that already exists
                                apiConsumer.addSubscription(apiTypeWrapper, userId, appId);
                            }
                    } else {
                            log.error("Failed to import Subscription as API/API Product " + name + "-" + version +
                            " as one or more tiers may be unavailable or the API/API Product may not have been published ");
                            skippedAPIList.add(subscribedAPI.getApiId());
                    }
```

Reply:
Fixed. Thanks for the input with ApiTypeWrapper.